src/Maestro/Maestro.Web.Tests/BuildController20200914Tests.cs

```diff
@@ -0,0 +1,140 @@
+                build.AzureDevOpsRepository.Should().Be(repository);
+                build.AzureDevOpsBranch.Should().Be(branch);
+            }
+
+            {
+                var resultCommit = await data.Controller.GetCommit(id);
+                var objResultCommit = (ObjectResult)resultCommit;
+                objResultCommit.StatusCode.Should().Be((int)HttpStatusCode.OK);
+                objResultCommit.Value.Should().BeAssignableTo<Commit>();
+                var commit = (Commit)objResultCommit.Value;
+
+                commit.Message.Should().Be(commitMessage);
+                commit.Sha.Should().Be(commitHash);
+                commit.Author.Should().Be(account);
+            }
+        }
+
+        private Task<TestData> BuildDefaultAsync()
+        {
+            return new TestDataBuilder().BuildAsync();
+        }
+
+        private sealed class TestDataBuilder
+        {
+            public async Task<TestData> BuildAsync()
+            {
+                string connectionString = await SharedData.Database.GetConnectionString();
+
+                ServiceCollection collection = new ServiceCollection();
+                collection.AddLogging(l => l.AddProvider(new NUnitLogger()));
+
+                var mockIRemoteFactory = new Mock<IRemoteFactory>();
+                var mockIRemote = new Mock<IRemote>();
+                mockIRemoteFactory.Setup(f => f.GetRemoteAsync(repository, It.IsAny<ILogger>())).Returns(Task.FromResult(mockIRemote.Object));
+                mockIRemote.Setup(f => f.GetCommitAsync(repository, commitHash)).Returns(Task.FromResult(new Microsoft.DotNet.DarcLib.Commit(account, commitHash, commitMessage)));
+
+                collection.AddSingleton(mockIRemote.Object);
+                collection.AddSingleton(mockIRemoteFactory.Object);
+                collection.AddSingleton(Mock.Of<IBackgroundQueue>());
+                collection.AddSingleton<IHostEnvironment>(new HostingEnvironment
+                {
+                    EnvironmentName = Environments.Development
+                });
+                collection.AddBuildAssetRegistry(options =>
+                {
+                    options.UseSqlServer(connectionString);
+                    options.EnableServiceProviderCaching(false);
+                });
+                collection.AddTransient<BuildsController>();
+                collection.AddSingleton<ISystemClock, TestClock>();
+                ServiceProvider provider = collection.BuildServiceProvider();
+
+                var controller = provider.GetRequiredService<BuildsController>();
+                var clock = (TestClock)provider.GetRequiredService<ISystemClock>();
+
+                return new TestData(provider, controller, clock);
+            }
+        }
+        private sealed class TestData : IDisposable
+        {
+            private readonly ServiceProvider _provider;
+            public BuildsController Controller { get; }
+            public TestClock Clock { get; }
```

Comment:
You don't appear to use the Clock in here, so you could remove this property. :-)